File: backend/Core/Domain/Entities/Car.cs
using Core.Domain.IdentityEntities;

namespace Core.Domain.Entities
{
    public class Car
    {
        public Guid Id { get; set; }
        public Guid UserId { get; set; }
        public Guid CarTopicId { get; set; }
        public string? OwnersDescription { get; set; }
        public ApplicationUser? User { get; set; }
        public CarTopic? CarTopic { get; set; }
        public ICollection<CarImage>? CarImages { get; set; }
        public RegistrationPlate? RegistrationPlate { get; set; }
        public CarDocuments? CarDocuments { get; set; }
    }
}

File: frontend/src/types/car.ts
import { CarDocuments } from "./carDocuments";
import { CarImage } from "./carImage";
import { CarTopic } from "./carTopic";
import { RegistrationPlate } from "./registrationPlate";

export interface Car {
  id: string;
  userId: string;
  carTopicId: string;
  ownersDescription: string;
  carDocuments: CarDocuments;
  registrationPlate: RegistrationPlate;
  carImages: CarImage[];
  carTopic: CarTopic;
}

export interface CarListItemDto {
  id: string;
  userId: string;
  carTopicId: string;
  ownersDescription: string;
  carDocuments: CarDocuments;
  registrationPlate: RegistrationPlate;
  carMainImage: CarImage;
  carTopic: CarTopic;
}

export interface CarFormData {
  carTopic: CarTopic;
  registrationCountry: string;
  registrationText: string;
  ownersDescription: string;
}